Documentation/pre-built-detection.md

```diff
@@ -0,0 +1,147 @@
+
+Deviation from this policy would result in higher maintanance cost for the source-build team, higher infrastructure load and bigger end product size.
+Nevertheless, if absolutely required, older versions of packages / libraries can be added to source-build throught a dedicated repository [source-build-reference-packages](https://github.com/dotnet/source-build-reference-packages). 
+
+To have a package added to the repo, please [log an issue](https://github.com/dotnet/source-build/issues/new/choose) explaining the use-case in question and the source-build team will help you make the right decision.
+
+If the package you require is already in the source-build reference packages, it can be directly added to the `Version.Details.xml` dependency list:
+
+```xml
+  <Dependency Name="Microsoft.SourceBuild.Intermediate.source-build-reference-packages" Version="x.y.z">
+    <Uri>https://github.com/dotnet/source-build-reference-packages</Uri>
+    <Sha>xxx000</Sha>
+    <SourceBuildTarball RepoName="source-build-reference-packages" ManagedOnly="true" />
+  </Dependency>
+```
+
+## Dependencies without any code
+
+Dependencies that do not have any code (eg. SDKs with just props and targets) can usually be added to the [text-only packages](https://github.com/dotnet/source-build-reference-packages/tree/main/src/textOnlyPackages/src). 
+If the package in question is not already present in the list, please [log an issue](https://github.com/dotnet/source-build/issues/new/choose) to determine if a text-only package would solve the use-case in question.
+
+To make the text-only packages available to the source-build of the repository, a dependency on source-build reference packages should be declared in the `Version.Details.props` file:
+
+```xml
+  <Dependency Name="Microsoft.SourceBuild.Intermediate.source-build-reference-packages" Version="x.y.z">
+    <Uri>https://github.com/dotnet/source-build-reference-packages</Uri>
+    <Sha>xxx000</Sha>
+    <SourceBuildTarball RepoName="source-build-reference-packages" ManagedOnly="true" />
+  </Dependency>
+```
+
+## External dependencies
+
+Some external dependencies, specifically those that have very few dependencies themselves and are simple to build, are placed in the [source-build-external](https://github.com/dotnet/source-build-externals) repository. 
+As with text-only packages, please [log an issue](https://github.com/dotnet/source-build/issues/new/choose) to get the process started.
+
+To make the text-only packages available to the source-build of the repository, a dependency on source-build externals package should be declared in the `Version.Details.props` file:
+
+```xml
+  <Dependency Name="Microsoft.SourceBuild.Intermediate.source-build-externals" Version="x.y.z">
+    <Uri>https://github.com/dotnet/source-build-externals</Uri>
+    <Sha>xxx000</Sha>
+    <SourceBuild RepoName="source-build-externals" ManagedOnly="true" />
+  </Dependency>
+```
+
+# FAQ
+
+WIP
+
+# Contacts
+
+WIP
```

Comment:
This should be @dotnet/source-build-internal.

Reply:
Thanks, added